### PR TITLE
black formatting and migrated to self.log logging in finetuning example

### PR DESCRIPTION
**Pull request by jspaezp**
## What does this PR do?

As discussed in #5174, it fixes the computer_vision_fine_tuning example.
 1.  It currently fails to run due to incorrect argument passing
 2.  Throws deprecation warning because it uses the old `return {'log': ...}` syntax instead of the current `self.log(...)` syntax.
 3.  Syntax does not follow the contributions guidelines (black formatting)

I noticed the changes seem fairly large due to black changing some of the formatting (which is suggested in the contribution guidelines pre-commit). Let me know if you want these changes to be rolled back to preserve the 'blame' of the repository

Let me know if you would like anything to be changed.

Fixes # (issue)

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? 
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃
I did have fun coding

Kindest wishes,
Sebastian

### CODE REVIEW

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 1]
```diff
@@ -166,23 +163,24 @@ class TransferLearningModel(pl.LightningModule):
...
+        lr: float = 1e-2,
+        lr_scheduler_gamma: float = 1e-1,
+        num_workers: int = 6,
+        **kwargs,
     ) -> None:
         """
         Args:
             dl_path: Path where the data will be downloaded
         """
-        super().__init__(**kwargs)
```
Comment (awaelchli) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:185]: that's a bit confusing. the lightning module does not accept any arguments, and there is no other superclass

**Review by awaelchli — APPROVED** [review]
LGTM, perhaps in a follow up one could further simplify by using the metrics api (Accuracy). 🥂

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 2]
```diff
@@ -364,75 +347,60 @@ def __dataloader(self, train):
         """Train/validation loaders."""
 
         _dataset = self.train_dataset if train else self.valid_dataset
-        loader = DataLoader(dataset=_dataset,
-                            batch_size=self.batch_size,
-                            num_workers=self.num_workers,
-                            shuffle=True if train else False)
+        loader = DataLoader(
+            dataset=_dataset, batch_size=self.batch_size, num_workers=self.num_workers, shuffle=True if train else False
```
Comment (rohitgr7) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:351]: maybe just rename the argument above to shuffle.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 3]
```diff
@@ -236,27 +232,24 @@ def train(self, mode=True):
...
         elif self.milestones[0] <= epoch < self.milestones[1] and mode:
             # Unfreeze last two layers of the feature extractor
-            freeze(module=self.feature_extractor,
-                   n=-2,
-                   train_bn=self.train_bn)
+            freeze(module=self.feature_extractor, n=-2, train_bn=self.train_bn)
 
     def on_epoch_start(self):
         """Use `on_epoch_start` to unfreeze layers progressively."""
         optimizer = self.trainer.optimizers[0]
```
Comment (rohitgr7) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:243]:
```suggestion
        optimizer = self.optimizers[0]
```

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 4]
```diff
@@ -271,25 +264,19 @@ def training_step(self, batch, batch_idx):
...
-        train_acc_mean = torch.stack([output['num_correct']
-                                      for output in outputs]).sum().float()
-        train_acc_mean /= (len(outputs) * self.batch_size)
-        return {'log': {'train_loss': train_loss_mean,
-                        'train_acc': train_acc_mean,
-                        'step': self.current_epoch}}
+        train_loss_mean = torch.stack([output["loss"] for output in outputs]).mean()
+        train_acc_mean = torch.stack([output["num_correct"] for output in outputs]).sum().float()
+        train_acc_mean /= len(outputs) * self.batch_size
+        self.log_dict({"train_loss": train_loss_mean, "train_acc": train_acc_mean, "step": self.current_epoch})
```
Comment (rohitgr7) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:279]: maybe use Accuracy from the metrics package?

File: pl_examples/domain_templates/computer_vision_fine_tuning.py [hunk 5]
```diff
@@ -303,59 +290,55 @@ def validation_step(self, batch, batch_idx):
...
+        val_loss_mean = torch.stack([output["val_loss"] for output in outputs]).mean()
+        val_acc_mean = torch.stack([output["num_correct"] for output in outputs]).sum().float()
+        val_acc_mean /= len(outputs) * self.batch_size
+        self.log_dict({"val_loss": val_loss_mean, "val_acc": val_acc_mean, "step": self.current_epoch})
 
     def configure_optimizers(self):
-        optimizer = optim.Adam(filter(lambda p: p.requires_grad,
-                                      self.parameters()),
-                               lr=self.lr)
+        optimizer = optim.Adam(filter(lambda p: p.requires_grad, self.parameters()), lr=self.lr)
```
Comment (rohitgr7) [inline pl_examples/domain_templates/computer_vision_fine_tuning.py:304]: if we just pass `self.parameters()` here, it won't update the ones with `requires_grad=False` right? so do we need to manually filter here?

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 1 above)
Comment (jspaezp) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:185]: I agree, It was odd to see it there and was the cause of an error that made the training fail as a whole; Since it passed an unused argument. The `kwargs` have been removed as part of this PR

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 2 above)
Comment (jspaezp) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:351]: I don't believe this would lead to more clarity, since setting `train` to `True` also changes the `dataset` to be used in the line above (and not only changes the shuffling).

Nontheless, `shuffle=True if train else False` could be replaced to a more simple `shuffle=train` and perhaps adding the type hints to make clear that `train` is expected to be a bool

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 4 above)
Comment (jspaezp) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:279]: This is a good idea, will go ahead and add/modify it.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 5 above)
Comment (jspaezp) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:304]: I am not sure about this particular one (neither do i know how to test it ....). I believe it is written that way to have the parameters added when the milestones are reached in `on_epoch_start`, by `_unfreeze_and_add_param_group`.

I will go ahead and remove it if you feel like it would still have the desired behavior.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 2 above)
Comment (rohitgr7) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:351]: ok cool. missed that. yeah, `shuffle=train` would be better.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 5 above)
Comment (rohitgr7) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py:304]: ok, let's leave it as it is. I didn't see that modified hook. Also, I think this might be better since Adam is adaptive and stores gradient info, so lesser the params, the lower will be the memory usage for grads.

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 4 above)
Comment (jspaezp) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py]: Done!

File: pl_examples/domain_templates/computer_vision_fine_tuning.py (hunk 2 above)
Comment (jspaezp) [reply pl_examples/domain_templates/computer_vision_fine_tuning.py]: Done in 73b34dd509c8f323ddf88b73457a22480ed99f55

**Review by rohitgr7 — APPROVED** [review]

**Review by SkafteNicki — APPROVED** [review]
LGTM
